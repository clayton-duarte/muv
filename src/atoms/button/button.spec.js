import React from 'react';

import Button from '.';

it('should render a button', () => {
  const wrapper = mount(
    <Button onClick={() => true} />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should render a anchor', () => {
  const wrapper = mount(
    <Button href="#" />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should change its style on warn', () => {
  const wrapper = mount(
    <Button warn href="#" />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should change its style on success', () => {
  const wrapper = mount(
    <Button success href="#" />,
  );
  expect(wrapper).toMatchSnapshot();
});


it('should change its style on outline', () => {
  const wrapper = mount(
    <Button outline href="#" />,
  );
  expect(wrapper).toMatchSnapshot();
});
