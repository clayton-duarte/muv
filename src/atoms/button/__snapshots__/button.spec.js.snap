// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change its style on outline 1`] = `
<Button
  disabled={false}
  href="#"
  onClick={null}
  outline={true}
  secondary={false}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem 0",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
>
  <styled.a
    disabled={false}
    href="#"
    onClick={null}
    outline={true}
    secondary={false}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem 0",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
  >
    <a
      className="sc-bwzfXH buCeHw"
      disabled={false}
      href="#"
      onClick={null}
    >
      default
    </a>
  </styled.a>
</Button>
`;

exports[`should change its style on success 1`] = `
<Button
  disabled={false}
  href="#"
  onClick={null}
  outline={false}
  secondary={false}
  success={true}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem 0",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
>
  <styled.a
    disabled={false}
    href="#"
    onClick={null}
    outline={false}
    secondary={false}
    success={true}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem 0",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
  >
    <a
      className="sc-bwzfXH eGVawg"
      disabled={false}
      href="#"
      onClick={null}
    >
      default
    </a>
  </styled.a>
</Button>
`;

exports[`should change its style on warn 1`] = `
<Button
  disabled={false}
  href="#"
  onClick={null}
  outline={false}
  secondary={false}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem 0",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
  warn={true}
>
  <styled.a
    disabled={false}
    href="#"
    onClick={null}
    outline={false}
    secondary={false}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem 0",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
    warn={true}
  >
    <a
      className="sc-bwzfXH ljeNxs"
      disabled={false}
      href="#"
      onClick={null}
    >
      default
    </a>
  </styled.a>
</Button>
`;

exports[`should render a anchor 1`] = `
<Button
  disabled={false}
  href="#"
  onClick={null}
  outline={false}
  secondary={false}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem 0",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
>
  <styled.a
    disabled={false}
    href="#"
    onClick={null}
    outline={false}
    secondary={false}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem 0",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
  >
    <a
      className="sc-bwzfXH ecSw"
      disabled={false}
      href="#"
      onClick={null}
    >
      default
    </a>
  </styled.a>
</Button>
`;

exports[`should render a button 1`] = `
<Button
  disabled={false}
  href={null}
  onClick={[Function]}
  outline={false}
  secondary={false}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem 0",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
>
  <styled.button
    disabled={false}
    href={null}
    onClick={[Function]}
    outline={false}
    secondary={false}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem 0",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
  >
    <button
      className="sc-bdVaJa bMyNl"
      disabled={false}
      href={null}
      onClick={[Function]}
    >
      default
    </button>
  </styled.button>
</Button>
`;
